src/main.cpp

```diff
@@ -300,6 +296,97 @@ CNodeState *State(NodeId pnode) {
     return &it->second;
 }
 
+class CMisbehaviorTracker {
+    typedef std::lock_guard<std::mutex> lock_guard;
+    struct Info {
+        const std::string name;
+        int badness;
+        bool ban;
+        Info(std::string&& nameIn, int badnessIn, bool banIn) : name(std::move(nameIn)), badness(badnessIn), ban(banIn) {}
```

Comment:
The `Info` ctor accepts an rvalue reference and binds it to `nameIn', making it an lvalue.